api/rpc_test.go

```diff
@@ -28,9 +29,21 @@ func TestRPCCallsUnderlyingNode(t *testing.T) {
 	t.Cleanup(cancel)
 	nd, server := setupNodeWithModifiedRPC(t)
 	url := nd.RPCServer.ListenAddr()
-	client, err := client.NewClient(context.Background(), "http://"+url)
-	t.Cleanup(client.Close)
+	// we need to run this a few times to prevent the race where the server is not yet started
+	var (
+		rpcClient *client.Client
+		err       error
+	)
+	for i := 0; i < 3; i++ {
```

Comment:
[non blocking]
This type of retry is really useful in testing. Since it seems like we don't have a helper function for this, we should create one like so, maybe in a `test` package:

```go

// Retry will call 'fn' 'tries' times, waiting 'durationBetweenAttempts'
// between each attempt, returning 'nil' the first time that 'fn' returns nil.
// If 'nil' is never returned, then the final error returned by 'fn' is
// returned.
func Retry(tries int, durationBetweenAttempts time.Duration, fn func() error) (err error) {
	for i := 1; i < tries; i++ {
		err = fn()
		if err == nil {
			return nil
		}
		time.Sleep(durationBetweenAttempts)
	}
	return fn()
}
```

Then this example would turn into:
```go
	err = test.Retry(3, time.Second, func() error {
		return client.NewClient(ctx, "http://"+url)
	})
	require.NoError(t, err)
```

Reply:
#1332